### ISSUE TITLE
cargo fix applies fixes multiple times on files imported multiple times

### ISSUE THREAD

**Issue by Marwes** [rust-lang/rust#57171]
cargo fix appears to try and apply its fixes multiple times if a file is used multiple times. This can be a problem when using "support" modules for tests as cargo fix can end up calculating the locations for the fixes on the original file for two test, apply them for the first test and then apply them for the second test, only the file has now changed and the fix gets applied in the wrong location. (At least that is what appears to happen).

```
tests/
    a.rs
    b.rs
    support/mod.rs # Used by a.rs and b.rs
```

Workaround is to specify `--jobs 1` when running `cargo fix`

**Comment by ehuss** [comment]
This should be fixed by rust-lang/cargo#6434 which will hit nightly once #56924 lands.

**Comment by jonas-schievink** [comment]
#56924 has been merged, closing this. @Marwes if this is still an issue, please leave a comment.
